src/neuroconv/tools/roiextractors/roiextractors.py

```diff
@@ -331,6 +343,54 @@ def data_generator(imaging):
     return nwbfile
 
 
+def _imaging_frames_to_hdmf_iterator(
+    imaging: ImagingExtractor,
+    iterator_type: Optional[str] = "v2",
+    iterator_options: Optional[dict] = None,
+):
+    """
+    Private auxiliary method to wrap frames from an ImagingExtractor into a DataChunkIterator.
+
+    Parameters
+    ----------
+    imaging : ImagingExtractor
+        The imaging extractor to get the data from.
+    iterator_type : str (optional, defaults to 'v2')
+        The type of iterator to use.
+        'v1' is the original DataChunkIterator of the hdmf data_utils.
+        https://hdmf.readthedocs.io/en/stable/hdmf.data_utils.html#hdmf.data_utils.DataChunkIterator
+        'v2' is the locally developed ImagingExtractorDataChunkIterator, which offers full control over chunking.
+    iterator_options : dict, optional
+        Dictionary of options for the iterator.
+        For 'v1' this is the same as the options for the DataChunkIterator.
+        For 'v2', see
+        https://hdmf.readthedocs.io/en/stable/hdmf.data_utils.html#hdmf.data_utils.GenericDataChunkIterator
+        for the full list of options.
+
+    Returns
+    -------
+    DataChunkIterator
+        The frames of the imaging extractor wrapped in an iterator object.
+    """
+
+    def data_generator(imaging):
+        for i in range(imaging.get_num_frames()):
+            yield imaging.get_frames(frame_idxs=[i]).squeeze().T
+
+    assert iterator_type in ["v1", "v2", None], "'iterator_type' must be either 'v1', 'v2' (recommended), or None."
+    iterator_options = dict() if iterator_options is None else iterator_options
+
+    if iterator_type is None:
+        return
```

Comment:
@CodyCBakerPhD 
If this returns None, then this will cause an error downstream in `hdmf` that we're trying to add data which is None. Should this be handled in `add_two_photon_series`?

Reply:
Oh, whoops, I meant to do something here lol

Reply:
Great, looks good to me.